### PR TITLE
[tests] fix blip2 edge case

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

(Carved from #40553, which is becoming messy)

Our tests have the `model.config.get_text_config(decoder=True).is_encoder_decoder` pattern, which doesn't make sense -- we're pulling the decoder if it exists, and then checking if it is encoder-decoder.

The pattern exists because of `blip2`, which wasn't setting `is_encoder_decoder` correctly -- if its LLM is an encoder-decoder model, then it is also an encoder-decoder model.

✅ blip 2 slow tests are passing

### CODE REVIEW

**Comment by gante** [comment]
run-slow: blip_2

File: setup.py [hunk 1]
```diff
@@ -79,14 +79,12 @@
...
     print(
-        (
-            "Warning: {} exists.\n\n"
-            "If you recently updated transformers to 3.0 or later, this is expected,\n"
-            "but it may prevent transformers from installing in editable mode.\n\n"
-            "This directory is automatically generated by Python's packaging tools.\n"
-            "I will remove it now.\n\n"
-            "See https://github.com/pypa/pip/issues/5466 for details.\n"
-        ).format(stale_egg_info)
+        f"Warning: {stale_egg_info} exists.\n\n"
```
Comment (gante) [inline setup.py:82]: make fixup corrected this 👀

File: tests/models/blip_2/test_modeling_blip_2.py [hunk 2]
```diff
@@ -1094,6 +1096,11 @@ def test_initialization(self):
     def test_internal_model_config_and_subconfig_are_same(self):
         pass
 
+    @parameterized.expand(TEST_EAGER_MATCHES_SDPA_INFERENCE_PARAMETERIZATION)
+    @unittest.skip("Won't fix: Blip2 + T5 backbone needs custom input preparation for this test")
```
Comment (gante) [inline tests/models/blip_2/test_modeling_blip_2.py:1100]: can be fixed, but I don't think it's worth it 👀

**Comment by gante** [comment]
run-slow: blip_2

File: tests/generation/test_utils.py [hunk 3]
```diff
@@ -400,7 +400,7 @@ def test_greedy_generate_dict_outputs_use_cache(self):
                 use_cache=True,  # Enable cache
             )
 
-            if model.config.get_text_config(decoder=True).is_encoder_decoder:
+            if model.config.is_encoder_decoder:
                 self.assertTrue(output_generate.sequences.shape[1] == self.max_new_tokens + 1)
```
Comment (zucchini-nlp) [inline tests/generation/test_utils.py:404]: cool thanks, can you check if Florence2 model works with these changes? It is same as blip and uses Bart as backbone

File: tests/models/blip_2/test_modeling_blip_2.py (hunk 2 above)
Comment (zucchini-nlp) [reply tests/models/blip_2/test_modeling_blip_2.py:1100]: agreed, let's assume that t5 passes the test so we are fine :)

File: tests/models/blip_2/test_modeling_blip_2.py [hunk 4]
```diff
@@ -1867,11 +1877,8 @@ def test_inference_t5_multi_accelerator(self):
         # Test output
         expected_ids_and_text = Expectations(
             {
-                ("cuda", None): ([0, 3, 7, 152, 67, 839, 1], "san diego"),
-                ("rocm", (9, 5)): (
-                    [0, 3, 7, 152, 2515, 11389, 3523, 1],
-                    "san francisco",  # TODO: check if this is ok
-                ),
+                ("cuda", None): ([0, 3, 7, 152, 2515, 11389, 3523, 1], "san francisco"),
```
Comment (zucchini-nlp) [inline tests/models/blip_2/test_modeling_blip_2.py:1882]: i believe the CUDA expectation incorrect from the beginning

**Review by zucchini-nlp — COMMENTED** [review]
Thanks, it is much better to have it in model config level. Can you also verify how Florence2 works with this PR? Those are the only two VLMs with encoder-decoder backbone afaik

File: tests/models/blip_2/test_modeling_blip_2.py (hunk 4 above)
Comment (gante) [reply tests/models/blip_2/test_modeling_blip_2.py:1882]: yes, I agree 👀

**Comment by gante** [comment]
run-slow: florence2

**Review by zucchini-nlp — APPROVED** [review]
Forgot to ✅

**Comment by gante** [comment]
@zucchini-nlp florence2 slow tests were mostly green (there were 3 FA2 failures, which are unrelated to this PR) ✅ 

In florence2, `is_encoder_decoder` was correctly set for the start 👍
